### ISSUE TITLE
[bitnami/kafka] How to incorporate ingress CNAME as valid bootstrap server for external advertised listener

### ISSUE THREAD

**Issue by 0music0** [bitnami/charts#8942]
### Name and Version

bitnami/kafka 15.0.4

### What steps will reproduce the bug?

1. In a custom kubernetes environment
2. I'm currently using externalAccess.service.type=NodePort to enable the default external listener port of 9094.  I then created a L7 Ingress load balancer with a private CNAME to point to the kafka-0-external service of port 9094.  However when using kafka-python consumer script externally to contact the service through the CNAME:80 or CNAME:443, the client script raised an unrecognized broker.  I also have tried having the CNAME point to a custom ingress service using Cluster IP to target port 9094 with the same labels as kafka-0-external, same issue.  I also have tried overwriting the KAFKA_ADVERTISED_LISTENER with  EXTERNAL://CNAME:80 and that does not work either.

### Are you using any custom parameters or values?

Here are the values with persistent storage parameters removed:

```
zookeeper.enabled=false
externalZookeeper.servers=zookeeper:2181
externalAccess.enabled=true
externalAccess.service.type=NodePort
externalAccess.autoDiscovery.enabled=true
serivceAccount.create=true
rbac.create=true
```

### What is the expected behavior?

Kafka python consumer script able to poll data from Kafka service successfully

### What do you see instead?

```
    raise Errors.UnrecognizedBrokerVersion()
kafka.errors.UnrecognizedBrokerVersion: UnrecognizedBrokerVersion
```

### Additional information

Strimzi.io's Kafka does support ingress for external listener but the architecture is too complex for my custom kubernetes environment today (https://strimzi.io/blog/2019/05/23/accessing-kafka-part-5/).  I'd like to know if there are existing methods of enabling such behavior with Bitnami Kafka.

**Comment by juan131** [comment]
Hi @0music0 

What authentication mechanism are you configuring in your external listener? Is it plaintext? Find more info below:

- https://github.com/bitnami/charts/blob/master/bitnami/kafka/values.yaml#L215

I saw you're using NodePort for the external service type + the auto-discovery method, are you obtaining the list of nodes in your cluster and using these IPs + the assigned node port to connect to your brokers? The NOTES displayed when installing the chart provides you the instructions to do so.

---

Note you might be interested on upgrading to Kafka 5.1.0 since this version includes support for configuring a different authentication mechanism for internal and external clients thanks to this contribution: 

- https://github.com/bitnami/charts/pull/8902

**Comment by 0music0** [comment]
Hi @juan131,

I am using plaintext authentication mechanism for the external listener.  My private cluster's external service configuration is limited to NodePort. I'm currently using the Cluster's IP address instead of the physical host's IP address provided through the NOTES section along with the NodePort number. However when accessing through my Ingress L7 CNAME using port 80, 443, and the NodePort number which points to 9094 via a global load balancer, Kafka Python was unable to recognize the address.  Is the recommendation to use SSL authentication with Kafka 5.1.0?

Thank you.

**Comment by juan131** [comment]
Hi @0music0 

Regarding your issue, I think this contribution by @fmulero is exactly what you're looking for:

- https://github.com/bitnami/charts/pull/8950

> Is the recommendation to use SSL authentication with Kafka 5.1.0?

If it's a production environment, it's definitely recommended to use SASL or SSL authentication specially if you're allowing connections from outside your K8s cluster.

**Comment by fmulero** [comment]
Hi 

We've just merged the changes to allow using names in external listeners. Two new parameters have been added:

* `externalAccess.service.loadBalancerNames`: It takes precedence over load balancer IPs, if loadBalancerNames is not defined then IP addresss will be used for external access in broker configuration. If this parameter is set, `externalAccess.service.loadBalancerIPs`is not mandatory. Length must be the same as replicaCount
* `externalAccess.service.loadBalancerAnnotations`: Array of load balancer annotations for external services. Useful to apply different annotations to each external load balancer. For example if external-dns is used to manage a DNS entry for the load balancer. Length must be the same as replicaCount.

The new chart version will be published in the repository during today
